File: apps/api/internal/category/categories.go
package categories

type ServiceCategory struct {
	Category  string               `json:"category"`
	Subcategories []ServiceSubCategory `json:"subcategories"`
}

type ServiceSubCategory struct {
	Id    uint8  `json:"id"`
	Title string `json:"title"`
}

var ServiceCategories []ServiceCategory = []ServiceCategory{
	{
		Category: "Home Services",
		Subcategories: []ServiceSubCategory{
			{Id: 1, Title: "Plumbing"},
			{Id: 2, Title: "Electrical work"},
			{Id: 3, Title: "Painting"},
			{Id: 4, Title: "Cleaning"},
			{Id: 5, Title: "Pest control"},
			{Id: 6, Title: "Landscaping"},
		},
	},
	{
		Category: "Professional Services",
		Subcategories: []ServiceSubCategory{
			{Id: 7, Title: "Legal advice"},
			{Id: 8, Title: "Accounting and financial services"},
			{Id: 9, Title: "Consulting"},
			{Id: 10, Title: "Marketing and advertising"},
			{Id: 11, Title: "Web development and IT services"},
		},
	},
	{
		Category: "Health and Wellness",
		Subcategories: []ServiceSubCategory{
			{Id: 12, Title: "Personal training"},
			{Id: 13, Title: "Nutrition consulting"},
			{Id: 14, Title: "Massage therapy"},
			{Id: 15, Title: "Counselling and therapy"},
			{Id: 16, Title: "Yoga and fitness classes"},
		},
	},
	{
		Category: "Education and Tutoring",
		Subcategories: []ServiceSubCategory{
			{Id: 17, Title: "Academic tutoring"},
			{Id: 18, Title: "Language lessons"},
			{Id: 19, Title: "Music and arts lessons"},
			{Id: 20, Title: "Test Preparation"},
			{Id: 21, Title: "Online courses and workshops"},
		},
	},
	{
		Category: "Events and Entertainment",
		Subcategories: []ServiceSubCategory{
			{Id: 22, Title: "Event planning and coordination"},
			{Id: 23, Title: "Photography and Videography"},
			{Id: 24, Title: "DJ and music services"},
			{Id: 25, Title: "Catering and food services"},
			{Id: 26, Title: "Venue booking"},
		},
	},
	{
		Category: "Automotive Services",
		Subcategories: []ServiceSubCategory{
			{Id: 27, Title: "Auto repair and maintenance"},
			{Id: 28, Title: "Car detailing"},
			{Id: 29, Title: "Towing and roadside assistance"},
			{Id: 30, Title: "Vehicle inspection"},
		},
	},
	{
		Category: "Beauty and Personal Care",
		Subcategories: []ServiceSubCategory{
			{Id: 31, Title: "Hairdressing and styling"},
			{Id: 32, Title: "Makeup and beauty services"},
			{Id: 33, Title: "Spa and wellness treatments"},
			{Id: 34, Title: "Nail care"},
		},
	},
	{
		Category: "Real Estate and Property Services",
		Subcategories: []ServiceSubCategory{
			{Id: 35, Title: "Property management"},
			{Id: 36, Title: "Home inspection"},
			{Id: 37, Title: "Real estate brokerage"},
			{Id: 38, Title: "Rental and leasing services"},
		},
	},
	{
		Category: "Technology and Digital Services",
		Subcategories: []ServiceSubCategory{
			{Id: 39, Title: "Software development"},
			{Id: 40, Title: "Graphic design"},
			{Id: 41, Title: "Digital Marketing"},
			{Id: 42, Title: "Social media management"},
		},
	},
	{
		Category: "Retail and Shopping Services",
		Subcategories: []ServiceSubCategory{
			{Id: 43, Title: "Personal shopping"},
			{Id: 44, Title: "Online shopping assistance"},
			{Id: 45, Title: "Product reviews and recommendations"},
		},
	},
}

File: web/src/lib/categories.ts
export type ServiceCategory = {
  category: string;
  subcategories: ServiceSubcategory[];
};

export type ServiceSubcategory = {
  id: number;
  title: string;
};

export const categories: ServiceCategory[] = [
  {
    category: 'Home Services',
    subcategories: [
      { id: 1, title: 'Plumbing' },
      { id: 2, title: 'Electrical work' },
      { id: 3, title: 'Painting' },
      { id: 4, title: 'Cleaning' },
      { id: 5, title: 'Pest control' },
      { id: 6, title: 'Landscaping' },
    ],
  },
  {
    category: 'Professional Services',
    subcategories: [
      { id: 7, title: 'Legal advice' },
      { id: 8, title: 'Accounting and financial services' },
      { id: 9, title: 'Consulting' },
      { id: 10, title: 'Marketing and advertising' },
      { id: 11, title: 'Web development and IT services' },
    ],
  },
  {
    category: 'Health and Wellness',
    subcategories: [
      { id: 12, title: 'Personal training' },
      { id: 13, title: 'Nutrition consulting' },
      { id: 14, title: 'Massage therapy' },
      { id: 15, title: 'Counselling and therapy' },
      { id: 16, title: 'Yoga and fitness classes' },
    ],
  },
  {
    category: 'Education and Tutoring',
    subcategories: [
      { id: 17, title: 'Academic tutoring' },
      { id: 18, title: 'Language lessons' },
      { id: 19, title: 'Music and arts lessons' },
      { id: 20, title: 'Test Preparation' },
      { id: 21, title: 'Online courses and workshops' },
    ],
  },
  {
    category: 'Events and Entertainment',
    subcategories: [
      { id: 22, title: 'Event planning and coordination' },
      { id: 23, title: 'Photography and Videography' },
      { id: 24, title: 'DJ and music services' },
      { id: 25, title: 'Catering and food services' },
      { id: 26, title: 'Venue booking' },
    ],
  },
  {
    category: 'Automotive Services',
    subcategories: [
      { id: 27, title: 'Auto repair and maintenance' },
      { id: 28, title: 'Car detailing' },
      { id: 29, title: 'Towing and roadside assistance' },
      { id: 30, title: 'Vehicle inspection' },
    ],
  },
  {
    category: 'Beauty and Personal Care',
    subcategories: [
      { id: 31, title: 'Hairdressing and styling' },
      { id: 32, title: 'Makeup and beauty services' },
      { id: 33, title: 'Spa and wellness treatments' },
      { id: 34, title: 'Nail care' },
    ],
  },
  {
    category: 'Real Estate and Property Services',
    subcategories: [
      { id: 35, title: 'Property management' },
      { id: 36, title: 'Home inspection' },
      { id: 37, title: 'Real estate brokerage' },
      { id: 38, title: 'Rental and leasing services' },
    ],
  },
  {
    category: 'Technology and Digital Services',
    subcategories: [
      { id: 39, title: 'Software development' },
      { id: 40, title: 'Graphic design' },
      { id: 41, title: 'Digital Marketing' },
      { id: 42, title: 'Social media management' },
    ],
  },
  {
    category: 'Retail and Shopping Services',
    subcategories: [
      { id: 43, title: 'Personal shopping' },
      { id: 44, title: 'Online shopping assistance' },
      { id: 45, title: 'Product reviews and recommendations' },
    ],
  },
];

export function getCategoryTitle(id: number): string {
  for (const category of categories) {
    for (const subcategory of category.subcategories) {
      if (subcategory.id === id) {
        return subcategory.title;
      }
    }
  }

  return '';
}
